### ISSUE TITLE
ClobBinding and BlobBinding shouldn't bind a null value on PreparedStatement::setClob and ::setBlob

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#14067]
jaybird throws an exception:

```
org.jooq.exception.DataAccessException: SQL [null]; Error while writing value at JDBC bind index: 2
	at org.jooq_3.18.0-SNAPSHOT.FIREBIRD.debug(Unknown Source)
	at org.jooq.impl.Tools.translate(Tools.java:3363)
	at org.jooq.impl.AbstractBindContext.bindValue(AbstractBindContext.java:77)
	at org.jooq.impl.Val.accept(Val.java:198)
	at org.jooq.impl.AbstractBindContext.bindInternal(AbstractBindContext.java:258)
	at org.jooq.impl.AbstractBindContext.visit0(AbstractBindContext.java:68)
	at org.jooq.impl.AbstractContext.visit(AbstractContext.java:346)
	at org.jooq.impl.QueryPartCollectionView.acceptElement(QueryPartCollectionView.java:252)
	at org.jooq.impl.QueryPartCollectionView.accept(QueryPartCollectionView.java:222)
	at org.jooq.impl.AbstractBindContext.bindInternal(AbstractBindContext.java:258)
	at org.jooq.impl.AbstractBindContext.visit0(AbstractBindContext.java:68)
	at org.jooq.impl.AbstractContext.visit(AbstractContext.java:346)
	at org.jooq.impl.AbstractQuery.execute(AbstractQuery.java:337)
	at org.jooq.impl.AbstractDelegatingQuery.execute(AbstractDelegatingQuery.java:115)
	at org.jooq.test.all.testcases.DataTypeTests.testLargeLOBsWithClobBinding0(DataTypeTests.java:1992)
	at org.jooq.test.all.testcases.DataTypeTests.testLargeLOBsWithClobBinding(DataTypeTests.java:1974)
	at org.jooq.test.jOOQAbstractTest.testLargeLOBsWithClobBinding(jOOQAbstractTest.java:4395)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.rules.TestWatcher$1.evaluate(TestWatcher.java:61)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:93)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:40)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:529)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:756)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:452)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
Caused by: java.sql.SQLException: Error while writing value at JDBC bind index: 2
	at org.jooq.impl.DefaultBindContext.bindValue0(DefaultBindContext.java:68)
	at org.jooq.impl.AbstractBindContext.bindValue(AbstractBindContext.java:74)
	... 46 more
Caused by: java.lang.NullPointerException: Cannot invoke "java.sql.Clob.getCharacterStream()" because "clob" is null
	at org.firebirdsql.jdbc.AbstractPreparedStatement.setClob(AbstractPreparedStatement.java:723)
	at org.jooq.tools.jdbc.DefaultPreparedStatement.setClob(DefaultPreparedStatement.java:244)
	at org.jooq.impl.ClobBinding.set(ClobBinding.java:98)
	at org.jooq.impl.DefaultBindContext.bindValue0(DefaultBindContext.java:63)
	... 47 more
```

We'll work around it by calling `PreparedStatement::setNull`, instead

**Comment by lukaseder** [comment]
Firebird bug here: https://github.com/FirebirdSQL/jaybird/issues/712

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.18.0, 3.17.5 (https://github.com/jOOQ/jOOQ/issues/14068), and 3.16.11 (https://github.com/jOOQ/jOOQ/issues/14069)
